interaction.go

```diff
@@ -0,0 +1,123 @@
+package discordgo
+
+import (
+	"time"
+)
+
+// InteractionDeadline is a deadline for responding to an interaction, if you haven't responded in the time, you won't be able to respond later.
+const InteractionDeadline = time.Second * 3
+
+// ApplicationCommand is representing application's slash command.
+type ApplicationCommand struct {
+	ID            string                      `json:"id,omitempty"`
+	ApplicationID string                      `json:"application_id,omitempty"`
+	Name          string                      `json:"name,omitempty"`
+	Description   string                      `json:"description,omitempty"`
+	Options       []*ApplicationCommandOption `json:"options,omitempty"`
+}
+
+// ApplicationCommandOptionType is type of an slash-command's option.
+type ApplicationCommandOptionType uint8
+
+// Application command option types.
+const (
+	_ = ApplicationCommandOptionType(iota)
```

Comment:
Consider doing `ApplicationCommandOptionSubCommand ApplicationCommandOptionType = iota + 1` instead of disposing values like this.